### ISSUE TITLE
What is the status of this project?

### ISSUE THREAD

**Issue by LennardF1989** [litedb-org/LiteDB#2230]
I'm a long time user and advocate of LiteDB where doing something like Lucene could be considered overkill.

However, I'm noticing a concerning lack of activity on this project. There are over 512 unhandled issues, last commit was 8 months ago, and 50 PR's are yet to be either accepted or rejected.

These three things are the first thing new users would look at to determine how safe a project would be to use in production. Currently these numbers are screaming "bail ship!".

So, what is the status? Is it still being worked on? Or should we start considering different solutions at this point?

**Comment by grofit** [comment]
The gitter is also dead, I have just picked this up for a prototype but there are a few barely documented features i.e `ITypeNameBinder` and some other bits, but gitter just seems to be people arriving, posting a question then leaving for past 12 months.

I appreciate the library is probably solid and works fine, but it does worry me that there is no centralised community to go to and ask questions when you are needing more complex use cases.

**Comment by perholmes** [comment]
I agree, and as much as I appreciate the effort that went into LiteDB, I'm terrified of using it for user-data in a long-term project. I'm going to bite the bullet and get SQLite running (this is Unity, so multi-platform SQLite isn't completely trivial).

I should say that I've had zero problems with LiteDB. But the V4/V5 feature split was a constant curve-ball, never knowing what version docs were for.

**Comment by perholmes** [comment]
If it helps anyone, we finally decided to abandon LiteDB in favor of SQLite. I'm just posting some experiences.

This is a Unity project, and I ended up going with SQLite4Unity (https://github.com/robertohuertasm/SQLite4Unity3d), which is based on SQLite-net (https://github.com/praeclarum/sqlite-net). Basically, SQLite-net is a wrapper for SQLite that finds the binaries on Windows, Linux and Android, and links to the native SQLite on Mac/iOS. Latest SQLite binaries can be downloaded from the SQLite project.

Our database layer is well abstracted, so it only took me about 5 hours to convert everything to SQLite. The ORM wrapper is very close to LiteDB in its implementation, and I actually found that using SQLite was cleaner than using LiteDB, where I sometimes had to open readers, get BsonDocuments, make them Enumerable, pick out fields, cast them from BsonValue etc. I was forced to go outside the ORM model in some LiteDB cases, and I could go clean ORM again with SQLite. All in all, I removed more code than I put in for the migration.

To be clear, I think that LiteDB is a fantastic idea. I really really wanted a source code to put straight into the project. But some of the files we create for our users may need to be read in 10 or 20 years, and I was simply too terrified of using LiteDB with an unclear commitment from the developers. We could risk having some corruption in the future and have basically no recourse for our users. Such corruption exists in tickets, with no resolution for years.

I really think that the LiteDB developers are making a terrible mistake by not expanding the team. I see highly competent people in the orbit of this project. I see pull requests that have been ignored for years. How is this possible?

I feel like the developers are saying that they would rather own and control the project than truly see it succeed. I can't square it any other way. I can't consider it a valid argument that qualified team members don't exist. For all its brilliance, it's still close-but-no-cigar, because the buy-in is not just for the current code, it's for the platform and its future and its team. I don't see much else than the current code, no team and no clear future. It's really too bad.

**Comment by tinmac** [comment]
It appears to me that the coms side of DbLite is flat & the developers are quiet, however [LiteDb-vNext](https://github.com/mbdavid/LiteDB-vNext) is under active development. It represents future features and improvements. Some clariry from @mbdavid wouldnt go amiss though.

**Comment by stigzler** [comment]
Agreed with above. It's so frustrating. I'm guessing that it's a small (possibly one man?) operation, but it's such a shame there's no community-driven support (e.g. discord or forums etc). I'd happily help out, but I'm a hobby coder, so wouldn't be the right lead for such a thing. It also worries me how this project evolves > V4 to V5 introduced significant changes and now there's a version called "vNext" which I'm guessing may be a re-write? Lats commit for this new version was a month ago at point of writing, so it's starting to feel like abandon ware. I do appreciate what the author has created here and know it's likely aside from his day job and hard to do everything, but some efforts towards communication and building a community may help. It's also quite unique + perfect for .net devs - I looked into mongodb which seems to need a server and Entity framework looks so inflexible.

**Comment by grofit** [comment]
The LiteDb vNext repository is archived as of 13th November.

It wasnt just a one man project to my knowledge, maybe it started that way but there was quite a few developers who joined the core team, I just think they have moved on and as the software is pretty stable there isnt much to do outside of adding new features etc, but as they dont seem to check here much the support/QA for the library is lacking, meaning even if it is stable no one new will be unable to get any guidance on gotchas/best practices or even simple problems etc.

**Comment by tinmac** [comment]
Problem is this thread isnt really representative.

Im defending @mbdavid as he's working on v6 & he's aiming to release this year.  Check out [this](https://github.com/mbdavid/LiteDB/issues/1300#issuecomment-1801404087) thread where he is there participating & giving us folk updates. Not dead to me.

**Comment by mbdavid** [comment]
Hello everybody!

I am still working on LiteDB, and this year more than ever. You can follow the evolution in the `dev` branch (or similar). I have updated the README with my upcoming tasks and progress of the new version.

But I would like to take advantage of this issue to also explain a little more about the LiteDB project.

I've been developing LiteDB since 2014 and I've always had it as a personal project that I love working on. For a long time I tried to help the community with its questions, bugs found and suggestions for improvements. I managed to maintain this for a while (in fact, for several years...), until I reached version 5. All my efforts were to stabilize the new version, improve documentation (as there was very little before) and continue helping the community.

The community grew and so did the demand for answers. I managed for a while, but there came a time when I received 5, 10 emails a day from github with any questions, bugs or suggestions for improvements, in addition to PRs, comments and questions on other networks (such as gitter and stackoverflow). A week out and I was no longer able to keep up with everything.

In 2019 I had to stop everything to deal with a particular situation and I was unable to return to the project. Even though I didn't know when I would be able to return, I didn't want to abandon the project. The demands on the community were huge and I simply couldn't serve everyone... so I ended up not serving anyone else.

During these years, I continued working in my company (which I have had since 2002). @kcsombrio is my partner in this company and @lbnascimento worked for a period on LiteB with me, but we had to allocate him to projects for our clients. @Feksa is currently our intern and is working on LiteDB with me.

LiteDB has always been open-source and I have never had any financial returns from it. Turning LiteDB into a product has always been my dream, as has my software factory company to prosper. Working with these two big dreams has never been easy, and Numeria has often been a priority (after all, it pays the bills :).

This year I managed to return to the project, and with that I reviewed a lot of things about LiteDB. The .NET platform has evolved a lot since 2017, when I started writing version 5.

So I decided to start version 6.0 - a rewrite of the code with a focus on simplicity and robustness, and of course - improving performance. I learned better about .NET memory management and the use of pointers and unmanaged structures. I improved the file format and how the information should be organized. I implemented a new, simpler and safer page management system - similar to those used in other databases, such as SQL Server. I rewrote the way expressions are resolved and the entire document structure system (BsonValue).

I really believe in this new version of LiteDB and I know that I can deliver a better database than the previous version. That's what makes perfect sense to me.

This new version is actually taking much longer than I imagined. I had to refactor many things several times until I arrived at simpler and safer solutions. I predicted that I would finish everything by October, but I couldn't. I am making my best efforts to deliver by the end of this year.

Finally, sorry for the distance and response time. I'm a very introverted person and that's why it's so difficult for me to communicate with so many people at the same time. I promise that I will continue coding... and responding to the community when I can :)

**Comment by grofit** [comment]
It shouldnt be just on you to manage this project and handle support queries, I think thats where a lot of the problem stems from, as Gitter isnt really fit for purpose these days, and there are not many people on it. So someone joins, asks a question, doesn't get an answer then leaves.

You could try spinning up a Discord server with community features which is what a lot of other popular open source projects do, this has a 2 fold benefit (and some problems).

- Its more used than Gitter
- You can more easily manage forum/thread style interactions on there
- More likely for people to have it turned on in background and see pings without needing to be fully active

The down sides of discord though are that you no longer have any way for web crawlers to index answers online so questions answered will generally only stay on there unless people take the time to replicate the really important Q/A things onto github readmes or SO self answered posts.

Ultimately an open source project lives or dies by its community, while a lot of people use this and the library is pretty stable and easy to work with, there is no real community which in turn forces more of a burden on you and will suck any time you have spare to work on the project away as you are having to help others.

Whatever you do, best of luck!

**Comment by JKamsker** [comment]
Thanks for asking, and for caring about the project. LiteDB is maintained by volunteers in their free time, and we do as much as we possibly can.

Development continues on the `dev` branch, which ships as the 6.x prereleases on NuGet. Please try the latest one (currently **6.0.0-prerelease.81**):

```
dotnet add package LiteDB --prerelease
```

If something doesn't work there, a new issue with a repro is the most useful way to help. Reviews and pull requests are very welcome too.

Closing this since the question is answered. If you'd like to discuss further, comment here and we'll reopen it.
